app/models/manageiq/providers/google/cloud_manager/event_catcher/stream.rb

```diff
@@ -0,0 +1,62 @@
+# Helper class responsible for polling a Google Pubsub topic and retreiving
+# event type messages. Parsing of the message is deferred to the caller of
+# #poll.
+class ManageIQ::Providers::Google::CloudManager::EventCatcher::Stream
+  def initialize(ems)
+    @ems = ems
```

Comment:
I think you can just `start` right away and set `@collecting_events` to true in the initializer. This also frees you from calling `start` in the Runner. Less methods, less errors :)